src/planner/ompl/CRRT.cpp

```diff
@@ -335,12 +335,24 @@ CRRT::Motion* CRRT::constrainedExtend(
   while (ptc == false)
   {
 
-    if (distToTarget == 0 || distToTarget - prevDistToTarget >= -mMinStepsize)
+    if (si_->equalStates(cmotion->state, gstate) || distToTarget - prevDistToTarget >= -mMinStepsize)
     {
       // reached target or not making progress
       break;
     }
 
+    // NOTE: (sniyaz) This check makes sure the resolution of the path on the
+    // constraint manifold and the resolution used for interpolation (*before*
+    // projection) do not differ wildly.
+    // TODO: Should we expose the factor to scale `mMaxStepsize` by?
+    double manifoldResolution = si_->distance(xstate, cmotion->state);
+    if (manifoldResolution > 2.0 * mMaxStepsize)
```

Comment:
Even if it isn't exposed, it might be better to pull it out of here and make it `constexpr double k<Name>` in the header or in the beginning of this file.

Reply:
Done!